### ISSUE TITLE
Qualified/namespaced maps should show up in references for that ns/keyword

### ISSUE THREAD

**Issue by Cyrik** [clojure-lsp/clojure-lsp#1369]
**Is your feature request related to a problem? Please describe.**
I'm often calling "find reference" on namespaced keywords/namespaces. This does not find reader-macro referenced maps like `#:some-ns{:a 1}` 

**Describe the solution you'd like**
Those occurrences should be returned from `textDocument/references` calls.

**Comment by ericdallo** [comment]
@Cyrik could you elaborate a little bit with a example? 
This seems to work for me:

![2022-11-10_12-14](https://user-images.githubusercontent.com/7820865/201133605-9c9366f6-1b32-4671-ad1b-533270d966d4.png)

**Comment by Cyrik** [comment]
Ah yes sorry, context...
I was actually talking about the `:foo` bit.
I was about to add that rename working would also be great, but I now realize that that's kind of a lot to ask if you rename one of the inner keys. I'd be happy if I got an error on rename, so I realize it would go wrong. Renaming a qualified key is actually how I ran into wanting this feature.

**Comment by ericdallo** [comment]
I can repro the issue with rename, which we can improve I think to rename the `#:foo` part when renaming `:foo/bar` -> `:bar/foo` for example, but I can't understand where the issue is in `textDocument/references`

**Comment by Cyrik** [comment]
`#:fo|o{:bar 1}` cursor at | produces =>
```
[Trace - 4:48:08 PM] Sending request 'textDocument/references - (27140)'.
Params: {
    "textDocument": {
        "uri": "file:///some-file.clj"
    },
    "position": {
        "line": 100,
        "character": 32
    },
    "context": {
        "includeDeclaration": true
    }
}


[Trace - 4:48:08 PM] Received response 'textDocument/references - (27140)' in 1ms.
Result: []
```
the same is true for ::foo if I had it referenced.
I think this should return either all instances for `#:foo` or even better all `:foo` qualified keywords.

**Comment by ericdallo** [comment]
yes, sounds like a new feature indeed, I never prioritized as it didn't seem that common to need to find all keywords of a namespace of namespaced keyword, the rename seems more important to fix though
